hazelcast/src/main/java/com/hazelcast/internal/util/VeryAbstractCompletableFuture.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util;
+
+import com.hazelcast.core.ExecutionCallback;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.util.ExceptionUtil;
+
+import java.util.concurrent.Executor;
+
+public abstract class VeryAbstractCompletableFuture<V> implements ICompletableFuture<V> {
```

Comment:
no JavaDoc? what's responsibility of this class? what it can do for me? when should I chose this one and when the other `AbstractCF`?